item-impl/src/test/scala/com/example/auction/item/impl/ItemEventProcessorSpec.scala

```diff
@@ -95,27 +95,29 @@ class ItemEventProcessorSpec extends AsyncWordSpec with BeforeAndAfterAll with M
       }
     }
 
-    "paginate item retrieval" in {
-      val creatorId = UUID.randomUUID
-      for {
-        _ <- Future.sequence(for (i <- 1 to 35) yield {
-          val item = sampleItem(creatorId).copy(title = s"title$i")
-          feed(item.id, ItemCreated(item))
-        })
-        items <- itemRepository.getItemsForUser(creatorId, api.ItemStatus.Created, 2, 10)
-      } yield {
-        items.count should ===(35)
-        items.page should ===(2)
-        items.pageSize should ===(10)
-        items.items should have size 10
-        items.items(4).title should ===("title11")
-      }
-
-    }
+    // TODO: Needs to be fixed according PagingState implementation because not make sense go to page 2 using this pattern
```

Comment:
Could it test from the beginning, ensure that the first result contains the expected number of items, and then that you can pass back the returned paging state to get more items?

It's a good idea to test the end of the results too (where the expected result count would be less than the maximum).

Reply:
Hi @TimMoore 
As you suggested, it was implemented a test case for pagination use case using `PagingState` serialized in string.